testing/upload_validation.ts

```diff
@@ -2160,6 +2162,12 @@ ${endpointKey ? 'endpointKey is set' : `requiresEndpointUrl is ${requiresEndpoin
     tools: z.array(toolSchema),
   });
 
+  const mcpProviderSchema = zodCompleteObject<McpProvider>({
+    endpoint: z.string().min(1).url('MCP provider endpoint must be a valid URL'),
+    agentName: z.string().min(1).max(Limits.BuildingBlockName),
+    agentVersion: z.string().min(1).max(50),
+  });
```

Comment:
I think this assumes the API OAuth endpoint is the same as the MCP OAuth endpoint - is that always true?  MCP requires PKCE I think, but our packs don't mandate it.  Might need to do more alignment work here.

Also - do we have to worry about dynamic client registration?  https://modelcontextprotocol.io/specification/2025-03-26/basic/authorization#dynamic-client-registration